src/hotspot/share/opto/loopopts.cpp

```diff
@@ -189,9 +189,7 @@ Node* PhaseIdealLoop::split_thru_phi(Node* n, Node* region, int policy) {
     IdealLoopTree *old_loop;
 
     if (x->is_Con()) {
-      // Constant's control is always root.
-      set_root_as_ctrl(x);
-      continue;
+      assert(get_ctrl(x) == C->root(), "constant control is not root");
```

Comment:
I think you should still execute `continue` here because we want to skip constants.

Reply:
Oops, that slipped through. Thanks! I fixed it.